internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -409,17 +411,42 @@ func (r *PreprovisioningImageReconciler) AddIronicAgentToInfraEnv(ctx context.Co
 		log.WithError(err).Error("failed to get corresponding infraEnv")
 		return false, err
 	}
+
+	// 1 - Check if agent image image is overriden
```

Comment:
Can we maybe put all of this tiered logic into a separate function?

Maybe `getIronicConfig` and return the new ironic config struct even if we get the data from one of the existing sources?